R/dummy_functions.R

```diff
@@ -92,23 +92,36 @@ example_module <- function(label = "example teal module", datanames = "all") {
   checkmate::assert_string(label)
   module(
     label,
+    ui = function(id) {
+      ns <- NS(id)
+      teal.widgets::standard_layout(
+        output = dataTableOutput(ns("table")),
+        encoding = div(
+          selectInput(ns("dataname"), "Choose a dataset", choices = NULL),
+          teal.widgets::verbatim_popup_ui(ns("rcode"), "Show R code")
+        )
+      )
+    },
     server = function(id, data) {
-      checkmate::assert_class(data, "tdata")
+      data_downgraded <- reactive(.tdata_downgrade(data))
+      checkmate::assert_class(data_downgraded(), "tdata")
       moduleServer(id, function(input, output, session) {
-        output$text <- renderPrint(data[[input$dataname]]())
+        output$text <- renderPrint(data_downgraded()[[input$dataname]]())
         teal.widgets::verbatim_popup_srv(
           id = "rcode",
-          verbatim_content = attr(data, "code")(),
+          verbatim_content = attr(data_downgraded(), "code")(),
           title = "Association Plot"
         )
       })
     },
     ui = function(id, data) {
+      data_downgraded <- reactive(.tdata_downgrade(data))
```

Comment:
There are two `ui`s. 

Another important thing:
- please don't (ever) send `data` to `ui`, this is one of the reasons we have a refactor.

Reply:
You will notice that it was `ui = function(id, data)` before this PR.

The two UI functions were a merge artefact.